### ISSUE TITLE
Default values for Non Null types

### ISSUE THREAD

**Issue by benwaffle** [graphql/graphql-js#1405]
When supplying a default value for a required field, it is not used and instead throws a validation error.

```js
const { graphql, buildSchema } = require('graphql')

const schema = buildSchema(`
type Query {
  hello(x: Int! = 42): Int
}
`)

const root = {
  hello: (args) => args.x
}

graphql(schema, '{ hello }', root).then(console.log)
``` 

Produces:
`GraphQLError: Field "hello" argument "x" of type "Int!" is required but not provided.`

**Comment by IvanGoncharov** [comment]
@benwaffle Should be fixed in #1274
Can you please try it on `14.0.0-rc.2`?

**Comment by benwaffle** [comment]
it works on 14.0.0-rc.2
